source/pdo_sqlsrv/pdo_dbh.cpp

```diff
@@ -362,34 +362,34 @@ const connection_option PDO_CONN_OPTS[] = {
 
 
 // close the connection
-int pdo_sqlsrv_dbh_close( pdo_dbh_t *dbh TSRMLS_DC );
+int pdo_sqlsrv_dbh_close( _Inout_ pdo_dbh_t *dbh TSRMLS_DC );
 
 // execute queries
-int pdo_sqlsrv_dbh_prepare( pdo_dbh_t *dbh, const char *sql,
-                            size_t sql_len, pdo_stmt_t *stmt, zval *driver_options TSRMLS_DC );
-zend_long pdo_sqlsrv_dbh_do( pdo_dbh_t *dbh, const char *sql, size_t sql_len TSRMLS_DC );
+int pdo_sqlsrv_dbh_prepare( _Inout_ pdo_dbh_t *dbh, _In_reads_(sql_len) const char *sql,
+                            _Inout_ size_t sql_len, _Inout_ pdo_stmt_t *stmt, _In_ zval *driver_options TSRMLS_DC );
+zend_long pdo_sqlsrv_dbh_do( _Inout_ pdo_dbh_t *dbh, _In_reads_bytes_(sql_len) const char *sql, _In_ size_t sql_len TSRMLS_DC );
 
 // transaction support functions
-int pdo_sqlsrv_dbh_commit( pdo_dbh_t *dbh TSRMLS_DC );
-int pdo_sqlsrv_dbh_begin( pdo_dbh_t *dbh TSRMLS_DC );
-int pdo_sqlsrv_dbh_rollback( pdo_dbh_t *dbh TSRMLS_DC );
+int pdo_sqlsrv_dbh_commit( _In_ pdo_dbh_t *dbh TSRMLS_DC );
```

Comment:
Should this not be inout instead, because sqlsrv_commit() modifies the connection handle?